src/Microsoft.DotNet.Build.Tasks/ExecWithRetries.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    /// <summary>
+    /// Run a command and retry if the exit code is not 0.
+    /// </summary>
+    public class ExecWithRetries : Microsoft.Build.Utilities.Task, ICancelableTask
```

Comment:
Is there any reason you cannot directly inherit from the Exec task directly? And in your execute you just wrap the execute of the Exec adding retry around it.